src/webOAuthServer.ts

```diff
@@ -193,10 +197,34 @@ export class WebOAuthServer extends AsyncCreatable<WebOAuthServer.Options> {
   protected async init(): Promise<void> {
     this.logger = await Logger.child(this.constructor.name);
     const port = await WebOAuthServer.determineOauthPort();
+    this.oauthConfig.loginUrl ??= AuthInfo.getDefaultInstanceUrl();
+    const env = new Env();
+
+    if (env.getBoolean('CODE_BUILDER')) {
+      if (this.oauthConfig.clientId && this.oauthConfig.clientId !== CODE_BUILDER_CONNECTED_APP_INFO.clientId) {
+        this.logger.warn(messages.getMessage('invalidClientId', [this.oauthConfig.clientId]));
+      }
+      this.oauthConfig.clientId = CODE_BUILDER_CONNECTED_APP_INFO.clientId;
+      const cbUri = env.getString('CODE_BUILDER_URI');
+      if (!cbUri) {
+        throw messages.createError('error.missingCodeBuilderUri');
+      }
+      const cbStateSha = env.getString('CODE_BUILDER_STATE_SHA');
+      if (!cbStateSha) {
+        throw messages.createError('error.missingCodeBuilderStateSha');
+      }
+      this.oauthConfig.state = JSON.stringify({
+        PORT: port,
+        CODE_BUILDER_URI: cbUri,
+        CODE_BUILDER_STATE_SHA: cbStateSha,
```

Comment:
I think we can slim this down a bit.

We can likely just look for and pass the encrypted state value and not worry about the `CODE_BUILDER_URI`

The state will contain either the CB URL or the instance id and we can redirect to the right place with either. Since its encrypted/decrypted on the server that should be enough to determine if the value is trust worthy.